dev-tools/reconfigurator-cli/tests/test_basic.rs

```diff
@@ -56,6 +58,24 @@ type ControlPlaneTestContext =
 #[nexus_test]
 async fn test_blueprint_edit(cptestctx: &ControlPlaneTestContext) {
     // Setup
+    //
+    // Add a zpool to both sleds, just to ensure that all new zones can find
+    // a transient filesystem wherever they end up being placed.
+    let _sled_agent_zpools = DiskTestBuilder::new(&cptestctx)
+        .on_sled(SledUuid::from_untyped_uuid(
+            cptestctx.sled_agent.sled_agent.id,
+        ))
+        .with_zpool_count(1)
+        .build()
+        .await;
+    let _sled_agent2_zpools = DiskTestBuilder::new(&cptestctx)
+        .on_sled(SledUuid::from_untyped_uuid(
+            cptestctx.sled_agent2.sled_agent.id,
+        ))
+        .with_zpool_count(1)
+        .build()
+        .await;
```

Comment:
This "works" -- it forces zpools to be allocated on all sleds before we try to create a blueprint -- but it's a little gross.

I've punted cleanup of the `DiskTest` stuff into https://github.com/oxidecomputer/omicron/pull/5939, just to keep this PR more reasonably-sized.